pybamm/models/full_battery_models/lithium_ion/mpm.py

```diff
@@ -0,0 +1,271 @@
+#
+# Many-Particle Model (MPM)
+#
+import pybamm
+from .base_lithium_ion_model import BaseModel
+
+
+class MPM(BaseModel):
```

Comment:
Excellent, I didn't think of this. I've now changed it to inherit from `SPM`. Annoyingly, one issue is that it can't use the `__init__` in `SPM` since the `MPM` needs to overwrite the citation `pybamm.citations.register("Marquis2019")`. Easily changed though, if you have a suggestion there.

Reply:
could do an if statement in the SPM like
```python3
if self.__class__ != "MPM":
```

Reply:
Great, added. I suppose in general you could add a method `set_citations` to the model, to be easily overwritten in any further subclass